tools/util/OperationManager.hpp

```diff
@@ -0,0 +1,1172 @@
+//////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC and Umpire
+// project contributors. See the COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (MIT)
+//////////////////////////////////////////////////////////////////////////////
+#ifndef REPLAY_OperationManager_HPP
+#define REPLAY_OperationManager_HPP
+
+#include <iostream>
+#include <cstdint>
+#include <vector>
+
+#include "umpire/Allocator.hpp"
+#include "umpire/strategy/AllocationAdvisor.hpp"
+#include "umpire/strategy/SizeLimiter.hpp"
+#include "umpire/strategy/MixedPool.hpp"
+#include "umpire/strategy/MonotonicAllocationStrategy.hpp"
+#include "umpire/strategy/SlotPool.hpp"
+#include "umpire/strategy/ThreadSafeAllocator.hpp"
+#include "umpire/ResourceManager.hpp"
+
+class Operation {
+public:
+ using AllocationOpMap = std::unordered_map<uint64_t, Operation*>;
+  std::function<void ()> op;
+
+  Operation(
+      std::vector<umpire::Allocator*>& alloc_array,
+      AllocationOpMap& alloc_operations
+  ) :   m_alloc_array(alloc_array)
+      , m_alloc_operations(alloc_operations)
+  {
+  }
+
+  void run()
+  {
+    op();
+  }
+
+  //
+  // AllocationAdvisor
+  //
+  template <typename... Args>
```

Comment:
Can the `bld_` functions  also be templated on the strategy type?

Reply:
Could these be renamed to e.g.`bldAdvisor` to follow the naming scheme?

Reply:
Please redefine the naming scheme for Umpire for me?  I forget.  Better yet, could this definition be added to the contribution page?

Reply:
Here's a brief breakdown: `class Something`, `void Something::doSomething()`, `Something::m_member`, `umpire::free_function()`

Reply:
And when do we use `Something::member` instead of `Something::m_member`?

Reply:
`Something::member` would be a function. The `m_` prefix is for class members, `s_` for statics.

Reply:
So items in `structs` follow a different convention?

[`struct AllocationRecord`](https://github.com/LLNL/Umpire/blob/develop/src/umpire/util/AllocationRecord.hpp#L22)

Reply:
AllocationRecord is a reserved name!